h2/src/main/org/h2/util/CacheLRU.java

```diff
@@ -51,11 +51,17 @@
         this.writer = writer;
         this.fifo = fifo;
         this.setMaxMemory(maxMemoryKb);
-        long tmpLen = MathUtils.nextPowerOf2(maxMemory / 64);
-        if (tmpLen > Integer.MAX_VALUE) {
-            throw new IllegalStateException("do not support this much cache memory: " + maxMemoryKb + "kb");
+        try {
+            // Since setMaxMemory() ensures that maxMemory is >=0,
+            // we don't have to worry about an underflow.
+            long tmpLen = maxMemory / 64;
+            if (tmpLen > Integer.MAX_VALUE) {
+                throw new IllegalArgumentException();
+            }
+            this.len = MathUtils.nextPowerOf2((int) tmpLen);
+        } catch (IllegalArgumentException e) {
+            throw new IllegalStateException("This much cache memory is not supported: " + maxMemoryKb + "kb", e);
```

Comment:
seems a little strange to throw and catch and then throw again like this?

Reply:
The try block is for the IllegalArgumentException that gets thrown by nextPowerOf2(). Explicitly throwing the IllegalArgumentException here is just to enter the same error path. I could copy the IllegalStateException from the catch block and throw it directly, but that would mean either duplicating the message, which would bring the danger of de-synchronization with future changes, or storing the error message in a variable, which seems like a waist, given that this should be a rare error case.

Do you have a better idea?

Reply:
ah, I did not see that nextPowerOf2() nows throws, in that case, the code is fine